model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -71,7 +71,14 @@ data class PackageConfiguration(
     }
 
     fun matches(otherId: Identifier, provenance: Provenance): Boolean {
-        if (id != otherId) return false
+        @Suppress("ComplexCondition")
+        if (!id.type.equals(otherId.type, ignoreCase = true) ||
+            id.namespace != otherId.namespace ||
+            id.name != otherId.name ||
+            id.version != otherId.version
```

Comment:
Would it make sense to extract this to a function similar to

https://github.com/oss-review-toolkit/ort/blob/35dde457188fd2e21c3cc1b979410cbd07beb21b/model/src/main/kotlin/PackageCuration.kt#L60-L63

just to get rid of the annotation?

Reply:
It would still have four conditions (the threshold is three) because it also compares the version which is handled separately in package curations because there it's not simply an equality check.
I also thought about writing it like below, but basically also only to avoid the warning:
```kotlin
if (!id.type.equals(otherId.type, ignoreCase = true)) return false
if (id.namespace != otherId.namespace) return false
...
```